packages/experiments/README.md

```diff
@@ -1,51 +1,90 @@
 # Experiments
 
-Private `__experimental` APIs that are not [exposed publicly plugin authors](https://make.wordpress.org/core/2022/08/10/proposal-stop-merging-experimental-apis-from-gutenberg-to-wordpress-core/#respond).
+`@wordpress/experiments`  enables sharing private `__experimental` APIs across `@wordpress` packages without
+[publicly exposing them to WordPress extenders](https://make.wordpress.org/core/2022/08/10/proposal-stop-merging-experimental-apis-from-gutenberg-to-wordpress-core/#respond).
 
-This package acts as a "dealer" that only allows WordPress packages to use the experimental APIs.
+## Getting started
 
-Each package needs to start by registering itself:
+Every `@wordpress` package wanting to privately access or expose experimental APIs must opt-in to `@wordpress/experiments`:
 
 ```js
-const { register } =
+// In packages/block-editor/experiments.js:
+import { __dangerousOptInToUnstableAPIsOnlyForCoreModules } from '@wordpress/experiments';
+export const { lock, unlock } =
 	__dangerousOptInToUnstableAPIsOnlyForCoreModules(
-		'<CONSENT STRING>',  // See index.js, this may change without notice.
-		'@wordpress/blocks'
+		'I know using unstable features means my plugin or theme will inevitably break on the next WordPress release.',
+		'@wordpress/block-editor' // The package opting in
 	);
 ```
 
-The function name communicates that plugins are not supposed to use it. To make double and triple sure, the first argument must be that exact consent string, and the second argument must be a known `@wordpress` package that hasn't opted in yet – otherwise an error is thrown.
+Each package may only opt in once. The function name communicates that plugins are not supposed to use it.
 
-Expose a new `__experimental` API as follows:
+The function will throw an error if the following conditions are not met:
+
+1. The first argument must exactly match the required consent string: `'I know using unstable features means my plugin or theme will inevitably break on the next WordPress release.'`. 
```

Comment:
❤️

Reply:
as much as I love this, I feel like we also need to be careful if we make this step to match that _promise_ with actually removing experimental interfaces and functions.

as someone recently pointed out in a DefinitelyTyped issue, the `__experimental` function arg has been in for a _long_ time, so continuing to call it experimental can feel ingenuine.

this came up with the original discussion about getting rid of experimental APIs. even if we say with our words that they will "quickly" disappear or break, at some point, if we have stabilized that code and built up a number of examples in Core of using them, then we have actually communicated that experimental code isn't, we've communicated something more like, "this stopped being experimental a while ago but we never renamed it"

want to make sure the severity of our words matches the integrity of our actions

Reply:
Agreed! While we seem to be stuck with certain deprecated and experimental APIs, e.g. editor.RichText is used by 255 plugins according to wpdirectory.net search, I hope that the ability to hide functions will catalyze adding and removing experiments at will.